### PR TITLE
fix php warning in php-fpm: "Unable to load dynamic library aerospike"

### PR DESCRIPTION
**Pull request by alchimik**
see also #300

### CODE REVIEW

File: php-fpm/Dockerfile-56 [hunk 1]
```diff
@@ -127,6 +127,10 @@ RUN if [ ${INSTALL_AEROSPIKE_EXTENSION} = true ]; then \
     && rm /tmp/aerospike-client-php.tar.gz \
 ;fi
 
+RUN if [ ${INSTALL_AEROSPIKE_EXTENSION} = false ]; then \
```
Comment (appleboy) [inline php-fpm/Dockerfile-56]: move the following

```
COPY ./aerospike.ini /usr/local/etc/php/conf.d/aerospike.ini
```

into the following block?

```
RUN if [ ${INSTALL_AEROSPIKE_EXTENSION} = true ]; then \
    ....
;fi
```

Comment (alchimik) [inline php-fpm/Dockerfile-56]: I don't know how to do it

Comment (appleboy) [inline php-fpm/Dockerfile-56]:
```diff
-COPY ./aerospike.ini /usr/local/etc/php/conf.d/aerospike.ini
RUN if [ ${INSTALL_AEROSPIKE_EXTENSION} = true ]; then \
+   COPY ./aerospike.ini /usr/local/etc/php/conf.d/aerospike.ini
    # Install the php aerospike extension
    curl -L -o /tmp/aerospike-client-php.tar.gz "https://github.com/luciano-jr/aerospike-client-php/archive/master.tar.gz" \
    && mkdir -p aerospike-client-php \
    && tar -C aerospike-client-php -zxvf /tmp/aerospike-client-php.tar.gz --strip 1 \
    && ( \
        cd aerospike-client-php/src/aerospike \
        && phpize \
        && ./build.sh \
        && make install \
    ) \
    && rm /tmp/aerospike-client-php.tar.gz \
;fi
```

So you don't need to add the following block

```
RUN if [ ${INSTALL_AEROSPIKE_EXTENSION} = false ]; then \
....
```

Does this resolve your problem.

Comment (alchimik) [inline php-fpm/Dockerfile-56]: COPY inside RUN? 
Got error - ```ERROR: Unknown instruction: CURL```


with backslash:
``` COPY ./aerospike.ini /usr/local/etc/php/conf.d/aerospike.ini \  ```
got - ```/bin/sh: 1: COPY: not found```

Comment (lightmania) [inline php-fpm/Dockerfile-56]: even by replacing COPY by the cp command, the result is still the same error

**Comment by alchimik** [comment]
What about this (like xdebug):
- remove "extension=aerospike.so" from aerospike.ini
- use  docker-php-ext-enable to enable extension
```
ARG INSTALL_AEROSPIKE_EXTENSION=false
ENV INSTALL_AEROSPIKE_EXTENSION ${INSTALL_AEROSPIKE_EXTENSION}
# Copy aerospike configration for remote debugging
COPY ./aerospike.ini /usr/local/etc/php/conf.d/aerospike.ini
RUN if [ ${INSTALL_AEROSPIKE_EXTENSION} = true ]; then \
    # Install the php aerospike extension
    curl -L -o /tmp/aerospike-client-php.tar.gz "https://github.com/luciano-jr/aerospike-client-php/archive/master.tar.gz" \
    && mkdir -p aerospike-client-php \
    && tar -C aerospike-client-php -zxvf /tmp/aerospike-client-php.tar.gz --strip 1 \
    && ( \
        cd aerospike-client-php/src/aerospike \
        && phpize \
        && ./build.sh \
        && make install \
    ) \
    && rm /tmp/aerospike-client-php.tar.gz \
    && docker-php-ext-enable aerospike \
;fi
```
- do not remove aerospike.ini if $INSTALL_AEROSPIKE_EXTENSION = false


same for opcache (see #617)

File: php-fpm/Dockerfile-56 (hunk 1 above)
Comment (appleboy) [inline php-fpm/Dockerfile-56]: Sorry for my mistake. You are right

**Pull request merged**